vendor/bat-native-ledger/src/bat/ledger/internal/publisher/server_publisher_fetcher.cc

```diff
@@ -0,0 +1,307 @@
+  DCHECK(ledger);
+  // NOTE: We are reusing publisher prefix list refresh interval for
+  // determining the cache lifetime of publisher details. At a later
+  // time we may want to introduce an additional option for this value.
+  return ledger->GetUint64Option(ledger::kOptionPublisherListRefreshInterval);
+}
+
+ledger::PublisherStatus PublisherStatusFromMessage(
+    const publishers_pb::ChannelResponse& response) {
+  switch (response.wallet_connected_state()) {
+    case publishers_pb::UPHOLD_ACCOUNT_KYC:
+      return ledger::PublisherStatus::VERIFIED;
+    case publishers_pb::UPHOLD_ACCOUNT_NO_KYC:
+      return ledger::PublisherStatus::CONNECTED;
+    default:
+      return ledger::PublisherStatus::NOT_VERIFIED;
+  }
+}
+
+ledger::PublisherBannerPtr PublisherBannerFromMessage(
+    const publishers_pb::SiteBannerDetails& banner_details) {
+  auto banner = ledger::PublisherBanner::New();
+
+  banner->title = banner_details.title();
+  banner->description = banner_details.description();
+
+  if (!banner_details.background_url().empty()) {
+    banner->background =
+        "chrome://rewards-image/" + banner_details.background_url();
+  }
+
+  if (!banner_details.logo_url().empty()) {
+    banner->logo = "chrome://rewards-image/" + banner_details.logo_url();
+  }
+
+  for (auto& amount : banner_details.donation_amounts()) {
+    banner->amounts.push_back(amount);
+  }
+
+  if (banner_details.has_social_links()) {
+    auto& links = banner_details.social_links();
+    if (!links.youtube().empty()) {
+      banner->links.insert(std::make_pair("youtube", links.youtube()));
+    }
+    if (!links.twitter().empty()) {
+      banner->links.insert(std::make_pair("twitter", links.twitter()));
+    }
+    if (!links.twitch().empty()) {
+      banner->links.insert(std::make_pair("twitch", links.twitch()));
+    }
+  }
+
+  return banner;
+}
+
+ledger::ServerPublisherInfoPtr ServerPublisherInfoFromMessage(
+    const publishers_pb::ChannelResponseList& message,
+    const std::string& expected_key) {
+  for (auto& entry : message.channel_responses()) {
+    if (entry.channel_identifier() != expected_key) {
+      continue;
+    }
+
+    // TODO(zenparsing): The previous JSON data had an "excluded"
+    // field, whereas the protobuf format does not. Do we still
+    // need this field?
+
+    auto server_info = ledger::ServerPublisherInfo::New();
+    server_info->publisher_key = entry.channel_identifier();
+    server_info->status = PublisherStatusFromMessage(entry);
+    server_info->address = entry.wallet_address();
+    server_info->updated_at =
+        static_cast<uint64_t>(base::Time::Now().ToDoubleT());
+
+    if (entry.has_site_banner_details()) {
+      server_info->banner =
+          PublisherBannerFromMessage(entry.site_banner_details());
+    }
+
+    return server_info;
+  }
+
+  return nullptr;
+}
+
+// TODO(zenparsing): This copied from components/brave_private_cdn.
+// Can we include it from bat-native-ledger?
```

Comment:
@NejcZdovc This functionality is duplicated from `components/brave_private_cdn`. Are we "allowed" to pull in that code?

Reply:
@bridiver @kylehickinson  wdyt? see any problems?

Reply:
I can't see where this was anymore (seems like the comment is gone?). Also I can't find a  `brave_private_cdn` in brave-core to check

Reply:
@kylehickinson [see here](https://github.com/brave/brave-core/blob/master/components/brave_private_cdn/private_cdn_helper.cc#L12) for the existing implementation.

The one in this PR is [here](https://github.com/brave/brave-core/pull/5568/files#diff-f53b2a5a835cc256a097da16fe4628e9R119). (I modified my version to avoid an unnecessary string copy.)

Reply:
@zenparsing Based on [its BUILD.gn file](https://github.com/brave/brave-core/blob/master/components/brave_private_cdn/BUILD.gn) I dont see any issue with adding it to iOS (We already have a dep on `//base` anyways)

Reply:
I'll need to create an overload of `RemovePadding` that takes a `StringPiece`; I'll use that code once that change is in place.